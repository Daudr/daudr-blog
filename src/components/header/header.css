.hero {
  width: 100%;
  min-height: 480px;
  height: 80vh;
  max-height: 560px;
  box-sizing: border-box;
  padding: 48px 48px 32px 48px;
  overflow: hidden;
  transform: skewY(8deg);
  transform-origin: 100%;

  position: absolute;
  top: 0;
  left: 0;

  z-index: -10;
}

@media (max-width: 480px) {
  .hero {
    max-height: 486px;
    padding-top: 40px;
    transform: none;
  }
}

.background-sky {
  background-color: #1976d2;
  background: linear-gradient(145deg, #000000, #636566);
  color: #fff;
}

h1.header-title,
h3.header-title {
  text-transform: uppercase;
  text-align: center;
  margin-bottom: 1.75rem;
}

* {
  font-family: 'Roboto', sans-serif
}
