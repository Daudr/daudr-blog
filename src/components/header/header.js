import React, { useEffect, useRef } from 'react'
import { Link } from 'gatsby'
import ReactFlagsSelect from 'react-flags-select'

import { scale, rhythm } from '../../utils/typography'

import 'react-flags-select/css/react-flags-select.css'
import './header.css'

export const Header = ({
  title,
  setSelectedLanguage,
  defaultLang,
  customSkyClass = ``,
}) => {
  const flagsRef = useRef(null)

  useEffect(() => {
    flagsRef.current.updateSelected(defaultLang)
  }, [defaultLang])

  const header = (
    <h1
      style={{
        ...scale(1.5),
        marginBottom: rhythm(1.5),
        marginTop: 0,
      }}
      className='header-title'
    >
      <Link
        style={{
          boxShadow: `none`,
          textDecoration: `none`,
          color: `inherit`,
        }}
        to={defaultLang === 'US' ? `/` : `/it`}
      >
        {title}
      </Link>
    </h1>
  )

  const selectLanguage = country => {
    const language = country === 'IT' ? 'it' : 'en'
    setSelectedLanguage(language)
  }

  return (
    <header
      style={{
        color: `white`,
      }}
    >
      <div className={`background-sky hero ${customSkyClass}`}></div>
      <div className='flag-select' style={{ position: `fixed`, right: `10px` }}>
        <ReactFlagsSelect
          defaultCountry={defaultLang}
          countries={['US', 'IT']}
          showOptionLabel={false}
          showSelectedLabel={false}
          onSelect={selectLanguage}
          placeholder=''
          ref={flagsRef}
        />
      </div>
      {header}
    </header>
  )
}

export default Header
