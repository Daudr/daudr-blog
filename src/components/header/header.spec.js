import React from 'react'
import renderer from 'react-test-renderer'

import { Header } from './header'

import { location } from '../../../__mocks__/location.mock'

describe('Header', () => {
	it('renders correctly', () => {
		const tree = renderer
			.create(<Header location={location} title={`Header Test`} />)
			.toJSON()
		expect(tree).toMatchSnapshot()
	})
})
