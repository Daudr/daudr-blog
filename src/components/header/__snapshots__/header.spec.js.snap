// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<header
  style={
    Object {
      "color": "white",
    }
  }
>
  <div
    className="background-sky hero"
  />
  <div
    className="flag-select"
    style={
      Object {
        "position": "fixed",
        "right": "10px",
      }
    }
  >
    <div
      className="flag-select "
    >
      <button
        aria-expanded={false}
        aria-haspopup="listbox"
        aria-labelledby="select_flag_button"
        className="flag-select__btn"
        disabled={false}
        id="select_flag_button"
        onClick={[Function]}
        onKeyUp={[Function]}
        style={
          Object {
            "fontSize": "16px",
          }
        }
        type="button"
      >
        <span
          className="flag-select__option flag-select__option--placeholder"
        >
          
        </span>
      </button>
    </div>
  </div>
  <h1
    className="header-title"
    style={
      Object {
        "fontSize": "3.95285rem",
        "lineHeight": "4.375rem",
        "marginBottom": "2.625rem",
        "marginTop": 0,
      }
    }
  >
    <a
      href="/it"
      style={
        Object {
          "boxShadow": "none",
          "color": "inherit",
          "textDecoration": "none",
        }
      }
    >
      Header Test
    </a>
  </h1>
</header>
`;
